Source/Core/loadKTX.js

```diff
@@ -0,0 +1,188 @@
+/*global define*/
+define([
+        '../ThirdParty/when',
+        './defined',
+        './DeveloperError',
+        './loadArrayBuffer',
+        './PixelFormat',
+        './RuntimeError'
+    ], function(
+        when,
+        defined,
+        DeveloperError,
+        loadArrayBuffer,
+        PixelFormat,
+        RuntimeError) {
+    'use strict';
+
+    var fileIdentifier = [0xAB, 0x4B, 0x54, 0x58, 0x20, 0x31, 0x31, 0xBB, 0x0D, 0x0A, 0x1A, 0x0A];
+    var endiannessTest = 0x04030201;
+
+    var sizeOfUint32 = 4;
+
+    function parseKTX(data) {
+        var byteBuffer = new Uint8Array(data);
+
+        var isKTX = true;
+        for (var i = 0; i < fileIdentifier.length; ++i) {
+            if (fileIdentifier[i] !== byteBuffer[i]) {
+                isKTX = false;
+                break;
+            }
+        }
+
+        if (!isKTX) {
+            throw new RuntimeError('Invalid KTX file.');
+        }
+
+        var view = new DataView(data);
+        var byteOffset = 12; // skip identifier
+
+        var endianness = view.getUint32(byteOffset, true);
+        byteOffset += sizeOfUint32;
+        if (endianness !== endiannessTest) {
+            // TODO: Switch endianness?
```

Comment:
I'm going to ignore this for now. I couldn't find a big endian file to test with without generating one from the same code. The KTX library has an issue to add tests as well: https://github.com/KhronosGroup/KTX/issues/26